extensions/sql-migration/src/wizard/createMigrationControllerDialog.ts

```diff
@@ -0,0 +1,485 @@
+		this.populateResourceGroups();
+	}
+
+	private async populateResourceGroups(): Promise<void> {
+		this.migrationControllerResourceGroupDropdown.loading = true;
+		let subscription = this._subscriptionMap.get((this.migrationControllerSubscriptionDropdown.value as azdata.CategoryValue).name)!;
+		const resourceGroups = await getResourceGroups(this.migrationStateModel.azureAccount, subscription);
+		let resourceGroupDropdownValues: azdata.CategoryValue[] = [];
+		if (resourceGroups && resourceGroups.length > 0) {
+			resourceGroups.forEach((resourceGroup) => {
+				resourceGroupDropdownValues.push({
+					name: resourceGroup.name,
+					displayName: resourceGroup.name
+				});
+			});
+		} else {
+			resourceGroupDropdownValues = [
+				{
+					displayName: constants.RESOURCE_GROUP_NOT_FOUND,
+					name: ''
+				}
+			];
+		}
+		this.migrationControllerResourceGroupDropdown.values = resourceGroupDropdownValues;
+		this.migrationControllerResourceGroupDropdown.loading = false;
+	}
+
+	private createControllerStatus(): azdata.FlexContainer {
+
+		const informationTextBox = this._view.modelBuilder.text().withProps({
+			value: constants.CONTROLLER_DIALOG_CONTROLLER_CONTAINER_DESCRIPTION
+		}).component();
+
+		const expressSetupTitle = this._view.modelBuilder.text().withProps({
+			value: constants.CONTROLLER_OPTION1_HEADING,
+			CSSStyles: {
+				'font-weight': 'bold'
+			}
+		}).component();
+
+		const expressSetupLink = this._view.modelBuilder.hyperlink().withProps({
+			label: constants.CONTROLLER_OPTION1_SETUP_LINK_TEXT,
+			url: ''
+		}).component();
+
+		expressSetupLink.onDidClick((e) => {
+			vscode.window.showInformationMessage(constants.FEATURE_NOT_AVAILABLE);
+		});
+
+		const manualSetupTitle = this._view.modelBuilder.text().withProps({
+			value: constants.CONTROLLER_OPTION2_HEADING,
+			CSSStyles: {
+				'font-weight': 'bold'
+			}
+		}).component();
+
+		const manualSetupButton = this._view.modelBuilder.hyperlink().withProps({
+			label: constants.CONTROLLER_OPTION2_STEP1,
+			url: 'https://www.microsoft.com/en-in/download/details.aspx?id=39717'
+		}).component();
+
+		const manualSetupSecondDescription = this._view.modelBuilder.text().withProps({
+			value: constants.CONTROLLER_OPTION2_STEP2
+		}).component();
+
+		const connectionStatusTitle = this._view.modelBuilder.text().withProps({
+			value: constants.CONTROLLER_CONNECTION_STATUS,
+			CSSStyles: {
+				'font-weight': 'bold'
+			}
+		}).component();
+
+		this._connectionStatus = this._view.modelBuilder.text().withProps({
+			value: ''
+		}).component();
+
+		const refreshButton = this._view.modelBuilder.button().withProps({
+			label: constants.REFRESH,
+		}).component();
+
+		const refreshLoadingIndicator = this._view.modelBuilder.loadingComponent().withProps({
+			loading: false
+		}).component();
+
+		refreshButton.onDidClick(async (e) => {
+			refreshLoadingIndicator.loading = true;
+			await this.refreshStatus();
+			refreshLoadingIndicator.loading = false;
```

Comment:
Make sure to set the state correctly here if an exception is thrown while refreshing